src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -462,5 +462,23 @@ public async Task DiscardDraftAsync(ContentItem contentItem)
                 _session.Save(publishedItem);
             }
         }
+
+        public async Task<ContentItem> CloneAsync(ContentItem contentItem)
+        {
+            var cloneContentItem = await NewAsync(contentItem.ContentType);
+            await CreateAsync(cloneContentItem, VersionOptions.Draft);
+
+            var context = new CloneContentContext(contentItem, cloneContentItem);
+
+            await Handlers.InvokeAsync(async handler => await handler.CloningAsync(context), _logger);
```

Comment:
Move after the item is cloned, right before `ClonedAsync`. The idea here is that we just want a way to provide two steps for events. There is nothing that can be done before it's cloned.

Reply:
Done